File: Semester4/Advanced Programming/Lab2_Hw/src/main/java/domain/entities/Artist.java
package domain.entities;

import domain.IHasId;

public class Artist implements IHasId<Integer> {
    private Integer artistId;
    private String artistName;


    /**
     * Constructor for Artist class
     * @param artistId - id of the artist
     * @param artistName - name of the artist
     */
    public Artist(Integer artistId, String artistName){
        this.artistId = artistId;
        this.artistName = artistName;
    }

    /**
     * Getters and setters
     */
    @Override
    public Integer getId() {
        return artistId;
    }

    @Override
    public void setId(Integer integer) {
        this.artistId = integer;
    }

    public String getArtistName() {
        return artistName;
    }

    public void setArtistName(String artistName) {
        this.artistName = artistName;
    }

    /**
     * Overriding equals method
     * @param ot - object to be compared
     * @return - true if the objects are equal, false otherwise
     */
    @Override
    public boolean equals(Object ot){
        if(ot == null){
            return false;
        }

        if(this == ot){
            return false;
        }

        if(this.getClass() != ot.getClass()){
            return false;
        }

        Artist art = (Artist)ot;
        return art.getId().equals(this.getId()) &&
               art.getArtistName().equals(this.getArtistName());
    }
}

File: Semester4/Advanced Programming/ThriftProj/C#Thrift/Lab2_Festival/Common/domain/entities/Artist.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Lab2_Festival.domain.entities
{
    public class Artist
    {
        /// <summary>
        /// Constructor for Artist class
        /// </summary>
        /// <param name="id">id of the artist</param>
        /// <param name="name">name of the artist</param>
        public Artist(int id, string name)
        {
            this.Id = id;
            this.ArtistName = name;
        }
        
        ///
        /// Gettesr and setters
        ///
        public int Id { get; set; }
        public string ArtistName { get; set; }

        /// <summary>
        /// Overriding equals method
        /// </summary>
        /// <param name="obj">object to be compared</param>
        /// <returns>true if the objects are equal, false otherwise</returns>
        public override bool Equals(object obj)
        {
            if (!(obj is Artist)) return false;
            var artist = (Artist)obj;
            return artist.Id == this.Id;
        }

        /// <summary>
        /// Overriding hashCode method
        /// </summary>
        /// <returns>integer representing the hashcode for the current element</returns>
        public override int GetHashCode()
        {
            var hashCode = 1183001917;
            hashCode = hashCode * -1521134295 + Id.GetHashCode();
            hashCode = hashCode * -1521134295 + EqualityComparer<string>.Default.GetHashCode(ArtistName);
            return hashCode;
        }
    }
}
